### ISSUE TITLE
Docker target handler should detect when the agent cannot be started

### ISSUE THREAD

**Issue by bourretp** [roboconf/roboconf-platform#389]
Issue #367 revealed that a the DM does not detect an error while starting the agent on a VM (cause was missing JVM). Result was that the VM keep staying in the `being deplyed` state... forever...

A target handler *must* detect a fault in the agent startup, otherwise DM is polluted with unreachable/non-existing VMs, that cannot be removed/undeployed.

With docker, things should be pretty easy, as we keep a hand on the `docker run` process, and could analyse its output. However I don't know if other targets will be able to do so, mainly because things are getting truly distributed, and the DM may not distinguish a looooooooong agent startup from a failed one. Maybe a timeout would help?

**Comment by vincent-zurczak** [comment]
Target handlers can already "detect" errors.
Not every possible error, though. However, target handlers should be able to terminate a machine no matter the agent's state. And this is already the case.

The original bug (#367) was already dedicated to Docker.
And the problem is in the Docker target only.

If no agent starts on an EC2 VM, the DM has no way to know about it.
And most likely, it will be a user mistake when he/she created the virtual image. We are not going to connect in SSH to check a process is running. And even if the agent is running, we are not going to diagnostic the messaging server/implementation and/or the firewall settings to understand why it fails. There is a debug API in the REST services of the DM. Getting a diagnostic annot be automatic. Docker is sort of an exception because it is local.

**Comment by bourretp** [comment]
Ok, renaming issue then...

**Comment by bourretp** [comment]
On a practical point of view, this issue looks difficult to resolve. Yes, Docker runs locally, but the way we use it, using its REST endpoint, make most of the operations asynchronous.

We could start the container, wait for a while, and check it is running, but the wait period is highly dependent of the container/image/host machine. So this is a non-solution (+ it's ugly).

The #390 issue may avoid the primary cause of the agent start problem (no Java installed), but other errors might stay undetected.

To sum up, this issue might become a WONT_FIX, but I'll have a fresher look on it later.
